Bluetooth: Controller: Fix scan window value corruption

cvinayak opened the pull request:
Fix scan window value corruption due to modification of stored ticks_window value when enabling continuous scan.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Fix scan window value corruption

Fix scan window value corruption due to modification of
stored ticks_window value when enabling continuous scan.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley commented on subsys/bluetooth/controller/ll_sw/ull_scan.c:
```diff
@@ -349,8 +350,9 @@ int ull_scan_reset(void)
 
-void ull_scan_params_set(struct lll_scan *lll, uint8_t type, uint16_t interval,
-			 uint16_t window, uint8_t filter_policy)
+uint32_t ull_scan_params_set(struct lll_scan *lll, uint8_t type,
+			     uint16_t interval, uint16_t window,
+			     uint8_t filter_policy)
```
Just FYI, the TSC changed the character limit to 100, removing the 80 recommendation

Thalley commented on tests/bluetooth/bsim_bt/bsim_test_mesh/overlay_low_lat.conf:
```diff
@@ -1,4 +1,3 @@
-CONFIG_BT_EXT_ADV=n
```
Why is extended advertising changed here?

cvinayak replied:
`CONFIG_BT_EXT_ADV` is default `n` and an explicit `n`  in `overlay_low_lat.conf` is inconsistent with tests not using this overlay.

PavelVPV replied:
It is y by default for mesh: https://github.com/zephyrproject-rtos/zephyr/blob/928e5bbcf08ccfe75f264d407e16fb1c7e82846e/subsys/bluetooth/Kconfig.adv#L10

cvinayak replied:
@PavelVPV  Any reason why it is `n` for nRF51 (7367980) ? If it is only used for multiple advertising sets, then it should work on nRF51 too, did it fail on target?

PavelVPV replied:
IIRC it didn't fit in flash or RAM.

cvinayak replied:
ok... I will go back to `CONFIG_BT_EXT_ADV=n` for nrf51 (low lat variant).  BTW, the `periodic_adv` sample builds, runs, is extended advertising and is discoverable on iPhone 13 Pro.

PavelVPV commented on tests/bluetooth/bsim_bt/bsim_test_mesh/prj.conf:
```diff
@@ -4,14 +4,18 @@ CONFIG_SYS_CLOCK_TICKS_PER_SEC=32768
-CONFIG_BT_BROADCASTER=y
-CONFIG_BT_TINYCRYPT_ECC=y
+CONFIG_BT_CTLR_ADVANCED_FEATURES=y
+CONFIG_BT_CTLR_SCAN_UNRESERVED=y
```
What is the purpose of explicitly enabling `CONFIG_BT_CTLR_SCAN_UNRESERVED` here? It is already enabled by default for mesh: https://github.com/zephyrproject-rtos/zephyr/blob/cccfc8fc8f627e127cf81b4960c3d2419a477e19/subsys/bluetooth/controller/Kconfig.ll_sw_split#L793-L796.

PavelVPV reviewed (changes requested):
Few comments from my side should be addressed.

cvinayak committed:
Bluetooth: Controller: Remove mesh low lat bsim test workaround

Remove mesh loopback group test workaround introduced in
commit 2a9cd8ffb1d3 ("Bluetooth: Controller: Add
EVENT_OVERHEAD_START_US when enabling states").

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented:
@PavelVPV and @Thalley This fix causes the friendship msg_mesh_low_lat bsim test to fail. Under low lat scan window uses time reservation which would cause any advertisement start to be skipped to next advertising interval. As the fix in this PR corrects the `ticks_window` value, and correct `ticks_slot` is reserved, hence failing the test.

I am thinking on, how to resolve this:
1. not have the msg_mesh_low_lat test
2. make mesh stack tolerant to delayed advertising start

PavelVPV committed:
tests: Bluetooth: Mesh: Send unseg msg to friend reliably

This is done to guarantee delivery of an unsegmented message from a
regular node to a friend node.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

PavelVPV commented:
> @PavelVPV and @Thalley This fix causes the friendship msg_mesh_low_lat bsim test to fail. Under low lat scan window uses time reservation which would cause any advertisement start to be skipped to next advertising interval. As the fix in this PR corrects the `ticks_window` value, and correct `ticks_slot` is reserved, hence failing the test.
> 
> I am thinking on, how to resolve this:
> 
> 1. not have the msg_mesh_low_lat test
> 2. make mesh stack tolerant to delayed advertising start

Could cherry-pick this patch: https://github.com/PavelVPV/zephyr/commit/939896b6df243e33304bf8ac173df8af39b5811d ?

cvinayak commented:
> Could cherry-pick this patch: [PavelVPV@939896b](https://github.com/PavelVPV/zephyr/commit/939896b6df243e33304bf8ac173df8af39b5811d) ?

Thank you very much, it helped.
